### PR TITLE
HSEARCH-4362 Customizable optional operators for the regexp query

### PR DESCRIPTION
**Pull request by fax4ever**
https://hibernate.atlassian.net/browse/HSEARCH-4362

### CODE REVIEW

**Comment by fax4ever** [comment]
Run `default|elasticsearch` build to test more ES versions: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-2820/2/pipeline

File: util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/backend/types/impl/StubIndexValueFieldType.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public void apply(StubIndexSchemaDataNode.Builder builder) {
 		public Builder(Class<F> valueClass) {
 			super( valueClass );
 			stubFactories(
-					new StubSearchPredicate.Factory(),
+					new StubSearchPredicate.RegexpFactory(),
 					PredicateTypeKeys.NESTED,
 					PredicateTypeKeys.MATCH,
 					PredicateTypeKeys.RANGE,
```
Comment (yrodiere) [inline util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/backend/types/impl/StubIndexValueFieldType.java:67]: You're assigning the regexp predicate factory to all sorts of predicates that are not the regexp predicate.

The right way to do what you want would be to remove PredicateTypeKeys.REGEXP from the list, and add a call to `stubFactories(new StubSearchPredicate.RegexpFactory(), PredicateTypeKeys.REGEXP)`.

Though TBH, this looks like dead code, so it's more for consistency than anything else.

**Review by yrodiere — CHANGES_REQUESTED** [review]
Thanks. Here are a few comments.

**Comment by fax4ever** [comment]
Thanks for the review @yrodiere!
I tried to address the requested changes.

**Review by yrodiere — APPROVED** [review]

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**